### PR TITLE
feat: support `pylint` check in github action

### LINKED ISSUE #315: enable python style check via pylint
## Expected Behavior

using Pylint for python code style check for pull requests

## Current Behavior

there is `pylintrc` already, but no pylint workflow

## Possible Solution

adding pylint check to `.github/workflows/style.yml`

## Notice

- there seems no reviewdog-pylint available, it's fine to run `pylint` directly
- the current code may generate many errors after enabling Pylint, you may skip specifying the directory for Pylint check in this issue (PRs to fix those error are also welcome)

### PR DESCRIPTION
**Pull request by mangoGoForward**
Signed-off-by: mango <xu.weiKyrie@foxmail.com>

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
Resolved #315 

* **What is the new behavior (if this is a feature change)?**
support `pylint` check in github action

### CODE REVIEW

**Review by tobegit3hub — COMMENTED** [review]
LGTM

**Review by tobegit3hub — APPROVED** [review]

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -48,3 +48,22 @@ jobs:
...
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pylint
+
+      - name: Analysing the code with pylint
+        run: |
+          find . -name '*.py' -exec pylint {} \;
```
Comment (aceforeverd) [inline .github/workflows/style.yml]: use `make pylint`

File: pylintrc [hunk 2]
```diff
@@ -146,6 +146,52 @@ disable=abstract-method,
         wrong-import-order,
         xrange-builtin,
         zip-builtin-not-iterating,
+        no-self-argument,
```
Comment (aceforeverd) [inline pylintrc]: do not change pylintrc. we may just allow pylint to fail, or ignore the fail status in workflow

**Review by aceforeverd — CHANGES_REQUESTED** [review]

File: .github/workflows/style.yml (hunk 1 above)
Comment (mangoGoForward) [reply .github/workflows/style.yml]: done.

File: pylintrc (hunk 2 above)
Comment (mangoGoForward) [reply pylintrc]: Done. reverted those changes

File: .github/workflows/style.yml [hunk 3]
```diff
@@ -48,3 +48,22 @@ jobs:
...
+          python-version: 3.8
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pylint
+
+      - name: Analysing the code with pylint
+        run: |
+          make pylint
```
Comment (aceforeverd) [inline .github/workflows/style.yml:71]: thank you. 
you can add a prop to this step to allow the job pass even on error, and comment something that should fix later, like:
```
# FIXME: fix the pylint errors in project and rm continue-on-error flag
continue-on-error: true
```

Comment (mangoGoForward) [reply .github/workflows/style.yml:71]: Done

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**